Quicksort.cpp

```diff
@@ -0,0 +1,84 @@
+#include <math.h>
+#include <stdio.h>
+#include <chrono>
+#include <iostream>
+#include <random>
+#include <vector>
+
+void swap(int* a, int* b)
+{
+    int t = *a;
+    *a = *b;
+    *b = t;
+}
+
+int partition(int arr[], int low, int high)
+{
+    int pivot = arr[high];    // pivot
+    int i = (low - 1);  // Index of smaller element
+
+    for (int j = low; j <= high - 1; j++)
+    {
+        // If current element is smaller than or
+        // equal to pivot
+        if (arr[j] <= pivot)
+        {
+            i++;    // increment index of smaller element
```

Comment:
Always try to use ++i unless you need i++
In this case it is not gonna make a difference because the compiler is smart enough.
For things more complex than an int you might do an unnecessary copy